src/vunnel/providers/mariner/parser.py

```diff
@@ -0,0 +1,196 @@
+            return None
+        return self.objects_by_id.get(test.object_value.object_ref, None)
+
+    def make_fixed_in(self, definition: Definition) -> FixedIn | None:
+        state = self.get_state(definition)
+        obj = self.get_object(definition)
+        if state is None or state.evr is None:
+            return None
+        if obj is None or obj.name is None:
+            return None
+        version = state.evr.value
+        if state.evr.operation == LTE:
+            version = "None"  # legacy API needs the string "None" instead of None
+        return FixedIn(Name=obj.name, NamespaceName=self.namespace_name(), VersionFormat="rpm", Version=version)
+
+    def vulnerability_id(self, definition: Definition) -> str | None:
+        if definition.metadata is None or definition.metadata.reference is None or definition.metadata.reference.ref_id is None:
+            return None
+        return definition.metadata.reference.ref_id
+
+    def description(self, definition: Definition) -> str | None:
+        if not definition.metadata:
+            return None
+        return definition.metadata.description
+
+    def vulnerabilities(self) -> Generator[Vulnerability, None, None]:
+        for d in self.definitions:
+            if d.metadata is None or d.metadata.severity is None:
+                self.logger.warning("skipping definition because severity could not be found")
+                continue
+            if d.metadata.description:
+                pass
+            link = ""
+            if d.metadata.reference and d.metadata.reference.ref_url:
+                link = d.metadata.reference.ref_url
+            fixed_in = self.make_fixed_in(d)
+            if not fixed_in:
+                continue
+            vulnerability_id = self.vulnerability_id(d)
+            if not vulnerability_id:
+                continue
+            yield Vulnerability(
+                Name=vulnerability_id,  # intentional; legacy API uses Name as field for vulnerability ID.
+                NamespaceName=self.namespace_name(),
+                Description=self.description(d) or "",
+                Severity=d.metadata.severity,
+                Link=link,
+                CVSS=[],  # NOTE: this is not present in OVAL XML.
+                # This is a list because because there are several versions
+                # Each version is a vector of info about the vulnerability
+                # serialized as a string like CVSS:3.1/AV:N/AC:L/PR:L/UI:N/S:U/C:H/I:H/A:H
+                # TODO: add test image to https://github.com/anchore/test-images
+                FixedIn=[fixed_in],
+                Metadata={},  # Ideally, there's no metadata here.
+            )
+
+
+MARINER_URL_BASE = "https://raw.githubusercontent.com/microsoft/CBL-MarinerVulnerabilityData/main/{}"
+MARINER_URL_FILENAME = "cbl-mariner-{}-oval.xml"
+
+
+class Parser:
+    logger = logging.getLogger("mariner-parser")
```

Comment:
this line is functional, but should probably be moved to within an instance instead of on the class definition. We've been deriving the logger from the class name if it isn't passed in :
https://github.com/anchore/vunnel/blob/bbe25d971eaf537864846494cfe7cce475ca0e00/src/vunnel/providers/nvd/manager.py#L24-L26

(Also note that you have a logger passed into the constructor but it is currently ignored)

Reply:
Should we just normalize towards "every class makes a logger based in `self.__class__.__name__`? Or are there times when we should be passing down?

Reply:
Right now the convention is to pass down from the provider, so everything is logged relative to that provider (and not necessarily the internal objects of that provider, which is an implementation detail and doesn't need to be exposed logging-wise). That being said, in case a logger isn't passed we go ahead at set a default one such that logging doesn't do something like raise an exception.

Reply:
Thanks! I'll push a rev that passes down the logger.